### PR TITLE
Make UY parser throw clearer error

### PR DESCRIPTION
**Pull request by pierresegonne**
## Issue

The `UY` data source is out, but the logs are quite cryptic about the error https://storage.googleapis.com/electricitymap-parser-logs/UY.html 

## Description

* Raise `ParserException` when the data to fetch the url can't be accessed. 

### Double check

- [x] I have tested my parser changes locally with `poetry run test_parser "zone_key"`
- [x] I have run `pnpx prettier --write .` and `poetry run format` to format my changes.

### CODE REVIEW

File: parsers/UY.py [hunk 1]
```diff
@@ -149,11 +153,10 @@ def fetch_consumption(
 ) -> list:
     """collects consumption data from ADME and format all data points for target_datetime"""
     if target_datetime is None:
-        target_datetime = arrow.utcnow().replace(tzinfo=UY_TZ).datetime
+        target_datetime = datetime.now().replace(tzinfo=UY_TZ)
```
Comment (pierresegonne) [inline parsers/UY.py:156]: Tried to get rid of arrow, but could not replace all of it, as I don't know what's the format of the string returned by the source where `arrow.get` is used

File: parsers/UY.py [hunk 2]
```diff
@@ -105,11 +110,10 @@ def fetch_production(
 ) -> list:
     """collects production data from ADME and format all data points for target_datetime"""
     if target_datetime is None:
-        target_datetime = arrow.utcnow().replace(tzinfo=UY_TZ).datetime
+        target_datetime = datetime.now().replace(tzinfo=UY_TZ)
```
Comment (VIKTORVAV99) [inline parsers/UY.py:113]:
```suggestion
        target_datetime = datetime.now(tz=UY_TZ)
```

File: parsers/UY.py (hunk 1 above)
Comment (VIKTORVAV99) [inline parsers/UY.py:156]:
```suggestion
        target_datetime = datetime.now(tz=UY_TZ)
```

File: parsers/UY.py [hunk 3]
```diff
@@ -184,11 +187,10 @@ def fetch_exchange(
 ) -> list:
     """collects exchanges data from ADME and format all data points for target_datetime"""
     if target_datetime is None:
-        target_datetime = arrow.utcnow().replace(tzinfo=UY_TZ).datetime
+        target_datetime = datetime.now().replace(tzinfo=UY_TZ)
```
Comment (VIKTORVAV99) [inline parsers/UY.py:190]:
```suggestion
        target_datetime = datetime.now(tz=UY_TZ)
```

**Review by VIKTORVAV99 — APPROVED** [review]
Just a suggestion around getting the datetime but nothing blocking.

You might also want raise an error on the initial URL with `if not r.ok`

File: parsers/UY.py (hunk 1 above)
Comment (VIKTORVAV99) [reply parsers/UY.py:156]: I'll be taking a look at our usage of arrow and removing it the next days so hopefully I can remove the last bits then.

Comment (pierresegonne) [reply parsers/UY.py]: Nice! But FYI here we won't be able to get rid of it until the source comes back online

**Pull request merged**